src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -547,7 +547,7 @@ private static function parseUrl(string $url, array $query = [], array $allowedS
             }
 
             // https://tools.ietf.org/html/rfc3986#section-3.3
-            $parts[$part] = preg_replace_callback("#[^-A-Za-z0-9._~!$&/'()[\]*+,;=:@\\\\^`{|}%]++#", function ($m) { return rawurlencode($m[0]); }, $parts[$part]);
+            $parts[$part] = preg_replace_callback("#[^-A-Za-z0-9._~!$&/'()[\]*+,;=:@\\\\^`|%]++#", function ($m) { return rawurlencode($m[0]); }, $parts[$part]);
```

Comment:
I'd keep this as is

Reply:
If I do, we're deviate from the RFC and the tests I've added to `Symfony\Component\HttpClient\Tests\HttpClientTraitTest::testResolveUrl` break. 

I think the `+` in this regular expression needed to be removed in #49580 too but haven't tested that.

Reply:
This regexp is to keep characters that are allowed in URLs unencoded. + is one of them. Encoding it would actually break URLs.